### ISSUE TITLE
Allow enity::$id to be something other than int

### ISSUE THREAD

**Issue by Keeehi** [nextras/orm#75]
Je omezující po entitě vyžadovat, aby měla id a navíc to bylo číslo.
1) Identifikátor nemusí být přece vůbec číselný.
2) Některé entity ani nemusejí mít identifikátor. (např. m:n relace rozšířená třeba o sloupce s časem vytvoření je dostatečně identifikovatelná cizími klíči a nepotřebuje svůj identifikátor)

Líbilo by se mi třeba řešení jako je v kdyby\doctrine kde Entita je v základu neidentifikovaná, pomocí traity z ní mohu udělat identifikovanou a nebo si to sám nadefinovat.

**Comment by hrach** [comment]
Actually, I don't know about any line o f code, which intentionally force id as int. It should work, but I have never tested. :angel: 

M:N relations needs their own identificator! Actually, ORM support composite primary key. (the support is not the best, see issues).

However, I can't imagine we could drop compulsory primary key, ORM wasn't designed int his in mind. Do you have some nice usecase?

**Comment by Keeehi** [comment]
Oh, I did not realize, that I can rewrite id in definition of my entity. So ok, I can force it for example to string.
But what if I don't want it to by named as id? My entity user can be identified by unique email address. I can make email property (primary) but the property id still hangs there. Sure, I don't have to use it but it is still visible in metadata in entity reflection. I can duplicate orm\Entity and create MyEntity and delete id property from it. But that breaks all orm because it assumes, that property id always exist even I don't use it. Is it possible to make this default identity implementation only optional?

> However, I can't imagine we could drop compulsory primary key, ORM wasn't designed int his in mind. Do you have some nice usecase?

I agree, that all entities need PK. I just thought that id property has to be always used because it is defined in Entity. Than I found out that is not true. What I had in mind was composite primary key from different properties than id. My bad.
